lib/base.js

```diff
@@ -244,33 +244,29 @@ class ConnectionPool extends EventEmitter {
     this._poolCreate().then((connection) => {
       debug('pool(%d): connected', IDS.get(this))
 
-      return this._poolDestroy(connection).then(() => {
-        if (!this._connecting) {
-          debug('pool(%d): not connecting, exiting silently (was close called before connection established?)', IDS.get(this))
-          return
-        }
+      this._poolDestroy(connection)
+      if (!this._connecting) {
+        debug('pool(%d): not connecting, exiting silently (was close called before connection established?)', IDS.get(this))
+        return
+      }
 
-        // prepare pool
-        this.pool = gp.createPool({
+      // prepare pool
+      this.pool = new tarn.Pool(
+        Object.assign({
           create: this._poolCreate.bind(this),
           validate: this._poolValidate.bind(this),
-          destroy: this._poolDestroy.bind(this)
-        }, Object.assign({
+          destroy: this._poolDestroy.bind(this),
           max: 10,
           min: 0,
-          evictionRunIntervalMillis: 1000,
           idleTimeoutMillis: 30000,
-          testOnBorrow: true
-        }, this.config.pool))
-
-        this.pool.on('factoryCreateError', this.emit.bind(this, 'error'))
```

Comment:
Removing this means we aren't getting the same errors being emitted that we used to

Reply:
There's just not a lot we can do about this because if we emit an error it crashes node...

Reply:
OK - fixed it as best as I could by adding a listenter to the `acquire` rejection